bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -535,6 +541,22 @@ func (r *KubeadmConfigReconciler) MachineToBootstrapMapFunc(o handler.MapObject)
 	return result
 }
 
+// MachinePoolToBootstrapMapFunc is a handler.ToRequestsFunc to be used to enqueue
+// request for reconciliation of KubeadmConfig.
+func (r *KubeadmConfigReconciler) MachinePoolToBootstrapMapFunc(o handler.MapObject) []ctrl.Request {
+	result := []ctrl.Request{}
+
+	m, ok := o.Object.(*clusterv1.MachinePool)
+	if !ok {
+		return nil
+	}
+	if m.Spec.Template.Spec.Bootstrap.ConfigRef != nil && m.Spec.Template.Spec.Bootstrap.ConfigRef.GroupVersionKind().GroupKind() == bootstrapv1.GroupVersion.WithKind("KubeadmConfig").GroupKind() {
```

Comment:
Based on other discussions we've had around handling/updating of versioning of objectReferences, I'm wondering if this should only compare group and kind, ignoring version?

@vincepri we'd also need to take that into account for other similar use cases such as the `MachineToBootstrapMapFunc` method above.

Reply:
I had that in one of my previous comments. Must have gotten lost w/the most recent push.

Reply:
The map functions I believe can only be registered and receive one specific version of an object, unless you meant something else?

Reply:
Oh what, that change did make it in here!

Reply:
This is less around the registering of the types for the watching and more about ignoring the api version of the remote object reference on the MachinePool resource that is passed into the map function.